script/tasks/unpauseAllDiamonds.ts

```diff
@@ -0,0 +1,198 @@
+
+          // get a valid nonce
+          const nextNonce = await getNextNonce(
+            pendingTransactions,
+            safeAddress,
+            network.name,
+            chain.id,
+            safeInfo.nonce
+          )
+
+          // prepare SAFE transaction
+          const safeTransaction = await safe.createTransaction({
+            transactions: [
+              {
+                to: diamondAddress as Address,
+                value: 0n,
+                data: calldata,
+                operation: OperationType.Call,
+                nonce: nextNonce,
+              },
+            ],
+          })
+
+          // sign transaction with SAFE_SIGNER_PRIVATE_KEY
+          const signedTx = await safe.signTransaction(safeTransaction)
+          const safeTxHash = await safe.getTransactionHash(safeTransaction)
+
+          // Store transaction proposal in MongoDB
+          try {
+            const result = await storeTransactionInMongoDB(
+              pendingTransactions,
+              safeAddress,
+              network.name,
+              chain.id,
+              signedTx,
+              safeTxHash,
+              senderAddress
+            )
+
+            if (!result.acknowledged) {
+              throw new Error(
+                `[${network.name}] MongoDB insert was not acknowledged`
+              )
+            }
+
+            consola.info('Transaction successfully stored in MongoDB')
+          } catch (error) {
+            consola.error('Failed to store transaction in MongoDB:', error)
+            throw error
+          }
+
+          consola.success(`[${network.name}] Transaction proposed`)
+        } catch (error) {
+          consola.error(
+            `[${network.name}] Error proposing unpause transaction:`,
+            error
+          )
+        }
+      })
+    )
+
+    await mongoClient.close()
+    consola.success('All networks processed successfully.')
+
+    process.exit(0)
+  },
+})
+
+async function getBlacklistedFacetAddresses(
+  networkName: string,
+  blacklistFacets: string
+): Promise<Address[]> {
+  // if blacklist is empty we dont need to look for addresses
+  if (!blacklistFacets) {
+    return []
+  }
+
+  // make sure that networkName is a valid supported chain
+  if (!isValidSupportedChain(networkName)) throw Error('Network ')
+
+  // Split the string into an array of facet names and trim whitespace
+  const facetNames = blacklistFacets.split(',').map((name) => name.trim())
+
+  // Retrieve the corresponding addresses for each facet name from the deploy log
+  const facetAddresses: Address[] = []
+  for (const facetName of facetNames) {
+    try {
+      const facetAddress = await getContractAddressForNetwork(
+        facetName,
+        networkName
+      )
+
+      if (!facetAddress)
+        throw new Error(`No ${facetName} address found in deploy log.`)
+
+      facetAddresses.push(facetAddress as Address)
+    } catch (error) {
+      consola.error(
```

Comment:
I think we should differentiate between the two types of errors that getContractAddressForNetwork can throw:
1. `throw Error(`Could not deploy log for network ${network} in ${environment}`)` - this is critical, as the deployment log is expected to exist. In this case, I suggest using consola.error() and rethrowing the error
2. `throw Error('Could not find address of contract ${contractName} for network ${network} in ${environment} deploy log')` - imo this is non-critical, it just means the contract name wasn’t found in the deployment log. Here, I’d recommend logging it with consola.info() and not rethrowing the error

Reply:
I kinda want to disagree. 
For me both errors are critical.

We want to disable a facet while unpausing a diamond, which itself is aa critical operation. 

If we do not find this address (doesnt matter for which reason imho) then we cannot continue as without this address we cannot prepare the calldata for the unpauseDiamond call. 
In that case I would like to know what failed so I can investigate this network and fix manually, then try again.

wdyt @mirooon

Reply:
I was thinking what if the blacklist includes facets that don’t exist in the deploy log? In that case, we could simply ignore them. But you’re right, even if those facets don’t exist, it still means the blacklist flag is incorrect (we throw an error) and the blacklist should be fixed during the second call.

thanks for the clarification @0xDEnYO